diff --git a/node_modules/ipfs-utils/src/http.js b/node_modules/ipfs-utils/src/http.js
index 59bc296..d13e825 100644
--- a/node_modules/ipfs-utils/src/http.js
+++ b/node_modules/ipfs-utils/src/http.js
@@ -130,7 +130,7 @@ class HTTP {
     // @ts-ignore
     const signal = anySignal([abortController.signal, opts.signal])
 
-    if (globalThis.ReadableStream != null && opts.body instanceof globalThis.ReadableStream && (isBrowser || isWebWorker)) {
+    if (typeof ReadableStream !== 'undefined' && opts.body instanceof ReadableStream && (isBrowser || isWebWorker)) {
       // https://bugzilla.mozilla.org/show_bug.cgi?id=1387483
       opts.body = new Blob(await all(browserReableStreamToIt(opts.body)))
     }
@@ -141,15 +141,10 @@ class HTTP {
       fetch(
         url.toString(),
         {
-          ...opts,
+          method: opts.method,
+          body: opts.body,
           signal,
-          // @ts-expect-error non-browser fetch implementations may take extra options
-          timeout: undefined,
           headers,
-
-          // https://fetch.spec.whatwg.org/#dom-requestinit-duplex
-          // https://github.com/whatwg/fetch/issues/1254
-          duplex: 'half'
         }
       ),
       opts.timeout,
